app/views/api/users.py

```diff
@@ -45,8 +47,24 @@ def get(self, args):
     @blp.response(201, UserSchema.Response)
     @blp.login_required
     def post(self, item):
-        """Add a new user"""
-        return UserSchema().post(item)
+        """Add a new user
+
+        The token is required to has registrations left and a room associated"""
+
+        db = current_app.session
+        token = db.query(Token).get(item['token_id'])
+
+        if token.registrations_left == 0:
+            abort(UnprocessableEntity, errors=dict(token_id='No registrations left for given token'))
```

Comment:
Do not forget, to adapt the 'errors' keyword here. Our custom abort does no longer have an 'errors' keyword.